funding/manager.go

```diff
@@ -2728,6 +2769,192 @@ func (f *Manager) waitForTimeout(completeChan *channeldb.OpenChannel,
+	//   output.
+	zeroScript := chainntnfs.ZeroTaprootPkScript.Script()
+	spendNtfn, err := f.cfg.Notifier.RegisterSpendNtfn(
+		&txIn.PreviousOutPoint, zeroScript,
+		broadcastHeight-fundingBroadcastBuffer,
+	)
+	if err != nil {
+		errChan <- err
+		return
+	}
+
+	defer spendNtfn.Cancel()
+
+	select {
+	case spend, ok := <-spendNtfn.Spend:
+		if !ok {
+			errChan <- fmt.Errorf("spend chan closed")
+			return
+		}
+
+		// If the spending transaction is the funding transaction,
+		// we'll send nil on errChan and exit.
+		if *spend.SpenderTxHash == fundingTxid {
+			errChan <- nil
+			return
+		}
+
+		// Before we register for a confirmation notification after
+		// conflictReorgBuffer blocks, we'll register for block
+		// notifications so that if conflictReorgCeiling blocks pass
+		// without us receiving the confirmation notification, we'll
+		// exit gracefully.
+		epochClient, err := f.cfg.Notifier.RegisterBlockEpochNtfn(nil)
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		defer epochClient.Cancel()
+
+		// We'll be immediately notified of the best block. This will
+		// inform our end height.
+		var bestHeight int32
+		select {
+		case epoch, ok := <-epochClient.Epochs:
+			if !ok {
+				errChan <- fmt.Errorf("epoch chan closed")
+				return
+			}
+
+			bestHeight = epoch.Height
+
+		case <-f.quit:
+			return
+
+		case <-doneChan:
+			return
+		}
+
+		// Set the ending height to conflictReorgCeiling blocks after
+		// the best height.
+		endHeight := bestHeight + conflictReorgCeiling
+
+		// Register for a confirmation notification of
+		// conflictReorgBuffer blocks for the conflict transaction.
+		// We'll choose the first output's pkScript to notify on since
+		// it doesn't matter which output we choose to notify on.
+		var (
+			conflictTxid       = spend.SpenderTxHash
+			conflictPkScript   = spend.SpendingTx.TxOut[0].PkScript
+			conflictHeightHint = spend.SpendingHeight
+		)
+
+		confNtfn, err := f.cfg.Notifier.RegisterConfirmationsNtfn(
+			conflictTxid, conflictPkScript, conflictReorgBuffer,
+			uint32(conflictHeightHint),
+		)
+		if err != nil {
+			errChan <- err
+			return
+		}
+
+		defer confNtfn.Cancel()
+
+		for {
+			select {
+			case _, ok := <-confNtfn.Confirmed:
+				if !ok {
+					err := fmt.Errorf("conf chan closed")
+					errChan <- err
+					return
+				}
+
+				// The conflict transaction is considered
+				// reorg-safe and we can signal the caller to
+				// clean up this channel.
+				errChan <- errFundingInputSpent
+				return
+
+			case epoch, ok := <-epochClient.Epochs:
+				if !ok {
+					err := fmt.Errorf("epoch chan closed")
+					errChan <- err
+					return
+				}
+
+				if epoch.Height >= endHeight {
+					// The conflict transaction did not
+					// reach a sufficient number of
+					// confirmations to be considered
+					// reorg-safe. It is possible that the
+					// conflict transaction was reorg'd
+					// out. In this case, we'll just fall
```

Comment:
I think the funding tx must have been confirmed at this point, given it's usually 6 and the reorg ceiling is 160. Having to consider reorg really makes things much more complicated...don't know an easy fix here, but maybe we could place the two confirmation ntfn in two goroutines, something like,
1. `RegisterSpendNtfn` for the inputs
2. upon receiving the spend notification, register two `RegisterConfirmationsNtfn`, one for the conflict, one for the funding tx
3. exit whenever receiving the confirm notification(but then we have to think about the relationship between the user configured conf target for funding tx and the reorg ceiling we choose here).

Just a thought, non-blocking.